torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -419,172 +448,186 @@ def step(self, closure: Optional[Callable[[], float]] = None, **kwargs: Any) ->
                         )
         _ = list(map(lambda x: x.wait(), handles))
 
-        # Sync hypothethical new results from the wrapped optimizer to the exposed param_groups
+        # Sync any updated attributes in the local optimizer to the exposed
+        # `param_groups`
         self._sync_param_groups(self.optim.param_groups, self.param_groups)
 
         return loss
 
     def load_state_dict(self, state_dict: Dict[str, Any]) -> None:
         r"""
-        Restore the global parameter groups as well as the shard.
+        Load the state pertaining to the given rank from the input
+        ``state_dict``, updating the local optimizer as needed.
 
         Arguments:
-            state_dict (dict): optimizer state. Should be an object returned
-                from a call to :meth:`state_dict`
+            state_dict (dict): optimizer state; should be an object returned
+                from a call to :meth:`state_dict`.
         """
-
-        for key, value in state_dict["state"].items():
-            param = self._index_to_param[key]
-
-            # Populate the sharded optimizer state on the fly
+        for index, value in state_dict["state"].items():
+            param = self._index_to_param[index]
             if self._param_to_rank[param] != self.rank:
-                state_dict["state"][key] = None
+                # Clear any state irrelevant to this rank
+                state_dict["state"][index] = None
             else:
+                # Load the parameter state to the local optimizer
                 self.optim.state[param] = _recursive_copy_to_device(value, non_blocking=True, device=param.device)
 
         super().load_state_dict(state_dict)
 
-        # Sync with the optimizer param groups
+        # Sync the input state with the exposed and local optimizer states
         self._sync_param_groups(state_dict["param_groups"], self.param_groups)
         self._sync_param_groups(self.param_groups, self.optim.param_groups)
 
     def local_state_dict(self) -> Dict:
         r"""
-        Gets this rank's ``state_dict``.
+        Gets this rank's local ``state_dict``.
 
         Returns:
-            The state of the optimizer as a :class:`dict`.
-            It contains two entries:
-
-            * state - a dict holding current optimization state. Its content
-                differs between optimizer classes.
-            * param_groups - a dict containing all parameter groups
+            The state of the local optimizer as a :class:`dict`, containing
+            two entries:
+
+            * state
+                a dict holding the current optimization state, where the exact
+                contents depends on the optimizer class.
+            * param_groups
```

Comment:
![Screen Shot 2021-06-20 at 9 02 40 PM](https://user-images.githubusercontent.com/31054793/122705612-f2350900-d20a-11eb-9c2b-45b9ed396059.png)
I changed it to be like this because any other text on the same line as `* state` or `* param_groups` appears in the grey box, but anything in the next line appears underneath in the white. The old way looked strange with `differs between optimizer classes.` being indented and in the white, while `* state - a dict holding current optimization state. Its content` was in the grey box above.